test/e2e/backup_test.go

```diff
@@ -0,0 +1,241 @@
+			Namespace: "openshift-etcd",
+		},
+		Spec: configv1alpha1.BackupSpec{
+			EtcdBackupSpec: configv1alpha1.EtcdBackupSpec{
+				Schedule: "* * * * *",
+				TimeZone: "UTC",
+				RetentionPolicy: configv1alpha1.RetentionPolicy{
+					RetentionType:   configv1alpha1.RetentionTypeNumber,
+					RetentionNumber: &configv1alpha1.RetentionNumberConfig{MaxNumberOfBackups: 5},
+				},
+				PVCName: pvcName,
+			},
+		},
+	}
+
+	_, err := configClient.Backups().Create(context.Background(), &backupCrd, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 15*time.Minute)
+	defer cancel()
+	err = wait.PollUntilContextCancel(ctx, 30*time.Second, false,
+		func(ctx context.Context) (done bool, err error) {
+			jobList, err := batchClient.Jobs(backupCrd.Namespace).List(context.Background(), metav1.ListOptions{})
+			if err != nil {
+				klog.Infof("error while getting job list: %v", err)
+				return false, nil
+			}
+
+			// the cronjob will keep at most 5 successful runs, we're waiting until the list has that many jobs completed
+			succeededJobs := 0
+			for _, j := range jobList.Items {
+				ownedByOurCron := false
+				for _, reference := range j.OwnerReferences {
+					if reference.Kind == "CronJob" && reference.Name == backupCrd.Name {
+						ownedByOurCron = true
+						break
+					}
+				}
+
+				if ownedByOurCron {
+					klog.Infof("Found job invocation: %s with status %s", j.Name, j.Status.String())
+					if j.Status.Succeeded > 0 {
+						succeededJobs++
+					}
+				}
+			}
+
+			if succeededJobs >= 5 {
+				klog.Infof("found enough succeeded jobs")
+				return true, nil
+			}
+
+			klog.Infof("found %d/5 succeeded jobs, retrying...", succeededJobs)
+			return false, nil
+		})
+	require.NoError(t, err)
+	// TODO(thomas): we need an easy way to "ls -l" the hostPath folder of all CP nodes, maybe oc debug?
+}
+
+func getOperatorClient(t *testing.T) *operatorversionedclient.Clientset {
+	kubeConfig, err := framework.NewClientConfigForTest("")
+	require.NoError(t, err)
+
+	operatorConfigClient, err := operatorversionedclient.NewForConfig(kubeConfig)
+	require.NoError(t, err)
+
+	return operatorConfigClient
+}
+
+func getConfigClient(t *testing.T) *configversionedclientv1alpha1.ConfigV1alpha1Client {
+	kubeConfig, err := framework.NewClientConfigForTest("")
+	require.NoError(t, err)
+
+	c, err := configversionedclientv1alpha1.NewForConfig(kubeConfig)
+	require.NoError(t, err)
+
+	return c
+}
+
+func getBatchClient(t *testing.T) *batchv1client.BatchV1Client {
+	kubeConfig, err := framework.NewClientConfigForTest("")
+	require.NoError(t, err)
+
+	c, err := batchv1client.NewForConfig(kubeConfig)
+	require.NoError(t, err)
+
+	return c
+}
+
+func ensureHostPathPVC(t *testing.T, pvcName string) {
+	kubeConfig, err := framework.NewClientConfigForTest("")
+	require.NoError(t, err)
+
+	c, err := corev1client.NewForConfig(kubeConfig)
+	require.NoError(t, err)
+
+	pv := v1.PersistentVolume{
```

Comment:
I cant get why we create the PV manually, isn't PVC creates a PV automatically as long as there is `StorageClass` ?

Reply:
dynamic provisioning doesn't work with hostPaths IIRC

Reply:
really :O .. how the hostpath CSI driver works !! .. let me check

Reply:
is there a CSI driver? I thought that's still hardcoded somewhere in kubelet

Reply:
yes u r right, it does not 👍🏽

Reply:
hahaha great, well it's a demo :)